app/js/user.js

```diff
@@ -88,6 +87,49 @@
           });
         });
       });
+    },
+
+    /**
+     * Updates the local user data as well as last sync time.
+     * @param {object} user_data - The ammended user data and settings i.e.
+     * user_data = {
+     *   user: user,
+     *   settings, settings
+     * };
+     * @returns The updated user object from indexedDB
+     */
+    update_user: function(user_data) {
+      SumoDB.update_user(user_data).then(function(user) {
+        user.last_sync = Date.now();
+        return asyncStorage.setItem('user', user).then(function() {
```

Comment:
Currently the user profile on the server does not have fields for the handset_type and operator. I am sending those fields, they just are not being stored at the moment. So, perhaps I need to first speak to @mythmon about adding those fields and close this pull request temporarily until that has landed.